awsclilinter/awsclilinter/cli.py

```diff
@@ -127,31 +202,80 @@ def main():
 
     script_content = script_path.read_text()
 
-    rules = [Base64BinaryFormatRule()]
-    linter = ScriptLinter(rules)
-    findings = linter.lint(script_content)
-
-    if not findings:
-        print("No issues found.")
-        return
+    rules = [Base64BinaryFormatRule(), PaginationRule()]
 
     if args.interactive:
-        findings = interactive_mode(findings, script_content)
-        if not findings:
+        # Do an initial parse-and-lint with all the rules simultaneously to compute the total
+        # number of findings for displaying progress in interactive mode.
```

Comment:
What's the value of displaying progress here? It seems a bit unnecessary to completely run through the entire set of possible changes only to do it again without just saving the diffs of what would have been changed and then accepting/denying them in the later loop. That would increase complexity of the solution, and may not matter for performance (scripts might not be that big).